winch/codegen/src/isa/aarch64/masm.rs

```diff
@@ -870,6 +870,16 @@ impl Masm for MacroAssembler {
         let _ = (context, kind);
         Err(anyhow!(CodeGenError::unimplemented_masm_instruction()))
     }
+
+    fn wasm_load_atomic(
+        &mut self,
+        _src: Self::Address,
+        _dst: WritableReg,
+        _size: OperandSize,
+        _sextend: Option<ExtendKind>,
+    ) -> Result<()> {
+        todo!()
```

Comment:
Could you return a proper `CodeGenError` instead of panicking? I'm in the process of enabling spec tests for aarch64, in general recoverable errors are preferred as it makes  it easier to define when a particular proposal is not fully supported in a particular backend (like threads in `aarch64`)

I believe we have `CodeGenError::unimplemented_masm_instruction()`